File: client/src/app/models/product.js
export default class Product {
    id = "";
    name = "";
    description = "";
    cost = "";
    brand = "";

    constructor(id, fullName) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.cost = cost;
        this.brand = brand;

    }

    static fromResponse(response) {
        return new Product(response.id, response.name, response.description, response.cost, response.brand);
    }
}

File: OnlineStore.Repository/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OnlineStore.Repository.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double Cost { get; set; }
        public string PictureUrl { get; set; }
        public int BrandId { get; set; }
        public Brand Brand { get; set; }
        public int CategoryId { get; set; }
        public Category Category { get; set; }

        public ICollection<CartProduct> CartProducts { get; set; }
    }
}
